From patchwork Sat Jan 30 15:56:13 2010
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [-stable,2.6.32.7] clocksource: fix compilation if no GENERIC_TIME
Date: Sat, 30 Jan 2010 15:56:13 -0000
From: Aaro Koskinen <aaro.koskinen@iki.fi>
X-Patchwork-Id: 75896

See http://patchwork.kernel.org/patch/75896/

Commit a9238ce3bb0fda6e760780b702c6cbd3793087d3 broke compilation on
platforms that do not implement GENERIC_TIME (e.g. iop32x):

  kernel/time/clocksource.c: In function 'clocksource_register':
  kernel/time/clocksource.c:556: error: implicit declaration of function 'clocksource_max_deferment'

Provide a dummy implementation of clocksource_max_deferment() for such
platforms.

Signed-off-by: Aaro Koskinen <aaro.koskinen@iki.fi>

---
kernel/time/clocksource.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/kernel/time/clocksource.c b/kernel/time/clocksource.c
index 5155dc3..7b3a9d0 100644
--- a/kernel/time/clocksource.c
+++ b/kernel/time/clocksource.c
@@ -502,6 +502,11 @@ static void clocksource_select(void)
 
 #else /* CONFIG_GENERIC_TIME */
 
+static inline u64 clocksource_max_deferment(struct clocksource *cs)
+{
+	return 0;
+}
+
 static inline void clocksource_select(void) { }
 
 #endif
